src/libmongoc/doc/debugging.rst

```diff
@@ -104,3 +104,9 @@ If you see "bson command installed by lldb_bson" at the beginning of your LLDB s
 
 .. _PyMongo: https://pypi.python.org/pypi/pymongo
 .. _install pip: https://pip.pypa.io/en/stable/installing/#installing-with-get-pip-py)
+
+Test assertions
+---------------
+
+To enable assertions that a mutex is locked upon entering functions that require the caller to lock the mutex, configure
```

Comment:
I would leave this vague, like `To enable runtime test assertions, configure with ...`. I foresee us adding more types assertions behind this flag.